functions/controllers/cruzpoints/index.js

```diff
@@ -0,0 +1,81 @@
+const functions = require("firebase-functions");
+const express = require("express");
+const cors = require("cors");
+const { jwtCheck, hasCreateAdmin, hasUpdateHacker } = require("../../utils/middleware");
+const { setDocument, queryDocument, transaction } = require("../../utils/database");
+const { customAlphabet } = require("nanoid");
+
+const nanoid = customAlphabet("ABCDEFGHIJKLMNOPQRSTUVWXYZ123456789");
+
+const cruzpoints = express();
+cruzpoints.disable("x-powered-by");
+cruzpoints.use(express.json());
+
+const auth0Config = functions.config().auth;
+const corsConfig = auth0Config ? auth0Config.cors : "";
+
+const corsOptions = {
+  origin: corsConfig,
+  optionsSuccessStatus: 200,
+};
+
+cruzpoints.use(cors(corsOptions));
+
+cruzpoints.post("/createActivity", jwtCheck, hasCreateAdmin, async (req, res) => {
+  try {
+    const code = nanoid(7);
```

Comment:
```suggestion
    const code = nanoid(7);
    if (!req.body.activity && !req.body.points)
      res.status(400).send({status: 400, message: "Missing Parameters in Request Body"});
```
We should validate that we're storing an activity and not `undefined`

Reply:
This is fine, but javascript will throw an error anyways when we try to access req.body.activity, will make the change though